<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1100</width>
    <height>900</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1100</width>
    <height>900</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Meshtastic Visualizer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>711</width>
      <height>41</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="notification_label">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>80</y>
      <width>921</width>
      <height>741</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_device">
     <attribute name="title">
      <string>Device</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>140</y>
        <width>461</width>
        <height>281</height>
       </rect>
      </property>
      <property name="title">
       <string>Channels</string>
      </property>
      <widget class="QTableWidget" name="channels_table">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>441</width>
         <height>241</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>90</y>
        <width>411</width>
        <height>611</height>
       </rect>
      </property>
      <property name="title">
       <string>Radio Output</string>
      </property>
      <widget class="QTextEdit" name="output_textedit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>391</width>
         <height>541</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="export_radio_button">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Export</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>391</width>
        <height>91</height>
       </rect>
      </property>
      <property name="title">
       <string>Connection</string>
      </property>
      <widget class="QPushButton" name="connect_button">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>30</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
      <widget class="QComboBox" name="device_combobox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>178</width>
         <height>25</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Serial port path</string>
       </property>
      </widget>
      <widget class="QPushButton" name="disconnect_button">
       <property name="geometry">
        <rect>
         <x>290</x>
         <y>30</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Disconnect</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="reset_nodedb_checkbox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>121</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Reset NodeDB</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_8">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>440</y>
        <width>461</width>
        <height>261</height>
       </rect>
      </property>
      <property name="title">
       <string>Device Info</string>
      </property>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Name:</string>
       </property>
      </widget>
      <widget class="QLabel" name="devicename_label">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>70</y>
         <width>131</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>71</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Bat Level: </string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>41</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Role:</string>
       </property>
      </widget>
      <widget class="QLabel" name="role_label">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>130</y>
         <width>111</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QProgressBar" name="batterylevel_progressbar">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>100</y>
         <width>71</width>
         <height>23</height>
        </rect>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>21</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Id:</string>
       </property>
      </widget>
      <widget class="QLabel" name="id_label">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>81</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>160</y>
         <width>71</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Hardware:</string>
       </property>
      </widget>
      <widget class="QLabel" name="hardware_label">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>160</y>
         <width>91</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_map">
     <attribute name="title">
      <string>Map</string>
     </attribute>
     <widget class="QPushButton" name="scan_button">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>50</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>Refresh map</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_7">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>881</width>
        <height>661</height>
       </rect>
      </property>
      <property name="title">
       <string>Map</string>
      </property>
      <widget class="QWebEngineView" name="nodes_map" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>861</width>
         <height>621</height>
        </rect>
       </property>
       <property name="url" stdset="0">
        <url>
         <string>about:blank</string>
        </url>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_nodes">
     <attribute name="title">
      <string>Nodes</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>631</width>
        <height>311</height>
       </rect>
      </property>
      <property name="title">
       <string>Mesh</string>
      </property>
      <widget class="QTableWidget" name="mesh_table">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>611</width>
         <height>271</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>10</y>
        <width>241</width>
        <height>191</height>
       </rect>
      </property>
      <property name="title">
       <string>Traceroute</string>
      </property>
      <widget class="QPushButton" name="traceroute_button">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>150</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Traceroute</string>
       </property>
      </widget>
      <widget class="QComboBox" name="tr_dest_combobox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>111</width>
         <height>25</height>
        </rect>
       </property>
       <property name="editable">
        <bool>false</bool>
       </property>
       <property name="currentText">
        <string/>
       </property>
       <property name="placeholderText">
        <string>Destination</string>
       </property>
      </widget>
      <widget class="QComboBox" name="tr_channel_combobox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>111</width>
         <height>25</height>
        </rect>
       </property>
       <property name="placeholderText">
        <string>Channel</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="tr_maxhops_spinbox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>111</width>
         <height>26</height>
        </rect>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
       <property name="value">
        <number>3</number>
       </property>
      </widget>
      <widget class="QTableWidget" name="traceroute_table">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>30</y>
         <width>101</width>
         <height>151</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="plot_groupbox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>330</y>
        <width>891</width>
        <height>371</height>
       </rect>
      </property>
      <property name="title">
       <string>Metrics</string>
      </property>
      <widget class="QWidget" name="horizontalLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>368</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Node</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="nm_node_combobox">
          <property name="editable">
           <bool>false</bool>
          </property>
          <property name="currentText">
           <string/>
          </property>
          <property name="placeholderText">
           <string>Node Id</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Metric</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="nm_metric_combobox">
          <property name="placeholderText">
           <string>Metric</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="nm_update_button">
          <property name="text">
           <string>Update</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="horizontalLayoutWidget_3">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>861</width>
         <height>281</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="plot_layout"/>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_messages">
     <attribute name="title">
      <string>Messages</string>
     </attribute>
     <widget class="QTextEdit" name="message_textedit">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>610</y>
        <width>421</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="send_button">
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>580</y>
        <width>51</width>
        <height>121</height>
       </rect>
      </property>
      <property name="text">
       <string>Send</string>
      </property>
     </widget>
     <widget class="QTableWidget" name="messages_table">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>10</y>
        <width>671</width>
        <height>551</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="export_chat_button">
      <property name="geometry">
       <rect>
        <x>810</x>
        <y>570</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Export</string>
      </property>
     </widget>
     <widget class="QListWidget" name="msg_channel_list">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>211</width>
        <height>151</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Channels</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Nodes</string>
      </property>
     </widget>
     <widget class="QListWidget" name="msg_node_list">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>211</width>
        <height>171</height>
       </rect>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget_4">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>570</y>
        <width>421</width>
        <height>31</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>On</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="msg_channel_label">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>To</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="msg_to_label">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Maximum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>103</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="remaining_chars_label">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QWebEngineView</class>
   <extends>QWidget</extends>
   <header>qwebengineview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
